### PR TITLE
Version creation and update form updates

### PR DESCRIPTION
**Pull request by agjohnson**
Fixes:

- #287
- #267

### CODE REVIEW

**Comment by humitos** [comment]
Note that when using "Fixes:"  with bullets, those issues are not closed automatically when the PR is merged, unfortunately 😞

File: src/js/project/index.js [hunk 1]
```diff
@@ -142,25 +146,82 @@ export class ProjectVersionCreateView {
...
-        // Required because the default of ``.ui.prompt`` is a rounded input
-        prompt: ".ui.text",
-      },
-      fields: {
-        title: "verbose_name",
-        description: "identifier",
-      },
+      error: errors,
       fullTextSearch: true,
+      maxResults: 30,
```
Comment (humitos) [inline src/js/project/index.js:157]: This should probably a variable shared with the number we inject in the URL as `limit=`

File: src/js/project/index.js [hunk 2]
```diff
@@ -142,25 +146,82 @@ export class ProjectVersionCreateView {
...
+                        ></i>
+                        ${when(
+                          version.active,
+                          () => html`
+                            <span class="ui horizontal label">
+                              <i class="fas fa-check icon"></i>
+                              Active
+                            </span>
+                          `,
+                        )}
```
Comment (humitos) [inline src/js/project/index.js:206]: Oh, I think this is the place where I'd like to change the order, not where I commented in the previous place.

**Review by humitos — COMMENTED** [review]
This is great!

My suggestion here is about style an rendering only, not functionality. I'd prefer to keep the order "icon + verbose name".

Also, I'd like to align the "Active" element to the right, as shown in https://github.com/readthedocs/ext-theme/issues/267#issuecomment-2065564928

For reference, this is how the current PR renders locally:

Screenshot_2024-04-22_13-09-24


I'm proposing something like this:

Screenshot_2024-04-22_13-12-00

This matches how we show the "Hidden" and "Default" version on the list of versions.

**Comment by agjohnson** [comment]
> Also, I'd like to align the "Active" element to the right

The reason I am not doing this is I am matching the project create search view. I decided we should not make a new pattern here.

image

There is an image to the right, we don't have an image for versions however.

>  I'd prefer to keep the order "icon + verbose name".

The reason I am not doing this is to match the project create search. Also because icon then title pushes the header over to the right, so the description portion is then too far left. This then also needs an icon to push the text right and there also isn't an icon that captures version.identifier -- which can be a commit, tag, or branch.

On the version detail page, I originally did this and it looked too noisy and cluttered. I reflected the change here, and this matches patterns used through this UI better now.

> This matches how we show the "Hidden" and "Default" version on the list of versions.

Indeed, but those are listing views. I'm aiming to consolidate the search results views for now.

File: readthedocsext/theme/templates/projects/version_form.html [hunk 3]
```diff
@@ -44,12 +44,11 @@ <h2 class="ui medium dividing header">
         <div class="ui fluid card">
           <div class="content">
             <div class="header">
-              <i class="fa-duotone {% if version.type == "branch" %}fa-code-branch {% elif version.type == "tag" %}fa-tag {% endif %}icon"></i>
-              <code>{{ version.verbose_name }}</code>
+              {{ version.verbose_name }}
+              <i class="fas {% if version.type == "branch" %}fa-code-branch{% elif version.type == "tag" %}fa-tag{% endif %} small icon"></i>
```
[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocsext/theme/templates/projects/version_form.html:48]: It's less aligned with the icon first, the description line doesn't line up with the title line in this example:



In both the search results and all of the listing views, objects are text and then supporting icons for object metadata, like project privacy level. We don't use icon and then text anywhere else, so I prefer to match the existing pattern. This UI was first touched before some of those patterns developed.

[the remark this answers is not in the document]
Comment (humitos) [reply readthedocsext/theme/templates/projects/version_form.html:48]: The screenshot doesn't load to me.

[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocsext/theme/templates/projects/version_form.html:48]: Odd. It was just your screenshot from above. Here's what I'm talking about however:

image

**Comment by humitos** [comment]
> There is an image to the right, we don't have an image for versions however.

Gotcha. Can we put the "Active" label in the middle of the row instead so it's aligned between all the versions as shown in the last screenshot of https://github.com/readthedocs/ext-theme/pull/337#pullrequestreview-2014445589? I think it looks lot better.

> Also because icon then title pushes the header over to the right, so the description portion is then too far left

Also note that header and description won't match anyways, since use different fonts (the one from the description is monospaced which make the text a little longer)

**Review by humitos — APPROVED** [review]

**Comment by agjohnson** [comment]
Also, perhaps another pattern that I might prefer here is expanding labels instead of text labels:

- https://github.com/readthedocs/ext-theme/issues/342

I would want to play around with this as an option here, though would be careful that an icon label was enough to signal to the user that the version is already active -- reducing the text to an icon might not be as clear.

**Comment by humitos** [comment]
Looks great! However, it seems pretty low priority to me right now. I'd leave it for when we are polishing the dashboard in the future. I'm fine with what you've done for now.

**Comment by agjohnson** [comment]
I was not planning on it, no. This is a feature addition for later.

**Pull request merged**